### ISSUE TITLE
CI Test Failure: Rate limit middleware unexpected 429 response

### ISSUE THREAD

**Issue by peterschutt** [litestar-org/litestar#1258]
```
_____________________________ test_exclude_opt_key _____________________________

    def test_exclude_opt_key() -> None:
        @get("/excluded", skip_rate_limiting=True)
        def handler() -> None:
            return None
    
        @get("/not-excluded")
        def handler2() -> None:
            return None
    
        config = RateLimitConfig(rate_limit=("second", 1), exclude_opt_key="skip_rate_limiting")
    
        with create_test_client(route_handlers=[handler, handler2], middleware=[config.middleware]) as client:
            response = client.get("/excluded")
            assert response.status_code == HTTP_200_OK
    
            response = client.get("/excluded")
            assert response.status_code == HTTP_200_OK
    
            response = client.get("/not-excluded")
            assert response.status_code == HTTP_200_OK
    
            response = client.get("/not-excluded")
>           assert response.status_code == HTTP_429_TOO_MANY_REQUESTS
E           assert 200 == 429
E            +  where 200 = <Response [200 OK]>.status_code

tests/middleware/test_rate_limit_middleware.py:132: AssertionError
=========================== short test summary info ============================
FAILED tests/middleware/test_rate_limit_middleware.py::test_exclude_opt_key - assert 200 == 429
 +  where 200 = <Response [200 OK]>.status_code
============ 1 failed, 2078 passed, 5 skipped, 1 xfailed in 48.59s =============
Error: Process completed with exit code 1.
```

<a href="https://polar.sh/litestar-org/litestar/issues/1258">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/litestar-org/litestar/issues/1258/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/litestar-org/litestar/issues/1258/pledge.svg">
</picture>
</a>

**Comment by provinzkraut** [comment]
I encountered this before. Seems to be flaky for some reason.

**Comment by peterschutt** [comment]
Don't think this should have been closed.

**Comment by provinzkraut** [comment]
We're using [time-machine](https://github.com/adamchainz/time-machine). If you want to attempt a fix based on that, go for it.

**Comment by peterschutt** [comment]
Check out the last couple of posts in [this thread](https://discord.com/channels/919193495116337154/1151927578370768958/threads/1179281714653184122) - on consecutive occasions when I encountered the error locally, the call to the `/not-excluded` route fell right on the second.

**Comment by peterschutt** [comment]
> So it confirms that timestamps are problem here. New second -> remove old cache objects -> no limit exceeds I will create PR.

thanks @kedod!

**Comment by peterschutt** [comment]
Closed in #3164
